# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux26/usb-hotfix.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Fixes certain USB 1 devices, such as from HP, Avision, et al.

  - Rene Rebe <rene@exactcode.de>

--- src.linux26.default.20070129.095929.2238.localhost/drivers/usb/host/uhci-q.c.orig	2006-11-29 22:57:37.000000000 +0100
+++ src.linux26.default.20070129.095929.2238.localhost/drivers/usb/host/uhci-q.c	2007-01-29 10:45:17.785970730 +0100
@@ -672,7 +672,7 @@
 	 * to do that is to put URBs on the low-speed queue while the device
 	 * isn't in the CONFIGURED state. */
 	if (urb->dev->speed == USB_SPEED_LOW ||
-			urb->dev->state != USB_STATE_CONFIGURED)
+			urb->dev->state != USB_STATE_DEFAULT)
 		qh->skel = uhci->skel_ls_control_qh;
 	else {
 		qh->skel = uhci->skel_fs_control_qh;
